source/dofs/dof_tools_constraints.cc

```diff
@@ -1070,6 +1101,1316 @@ namespace DoFTools
+              unsigned int next_index = 0;
+
+              // DoFs on vertices:
+              // Nedelec elements have no DoFs on the vertices
+
+              // DoFs on lines:
+              for (unsigned int line = 0;
+                   line < GeometryInfo<dim>::lines_per_face;
+                   ++line)
+                for (unsigned int dof = 0; dof != fe.n_dofs_per_line(); ++dof)
+                  dofs_on_mother[next_index++] =
+                    this_face->line(line)->dof_index(dof, fe_index);
+
+              // DoFs on the face:
+              for (unsigned int dof = 0; dof != fe.n_dofs_per_quad(face); ++dof)
+                dofs_on_mother[next_index++] =
+                  this_face->dof_index(dof, fe_index);
+
+              // check that we have added all DoFs
+              AssertDimension(next_index, dofs_on_mother.size());
+
+              // fill the DoF on the children:
+              Assert(dof_handler.get_triangulation()
+                         .get_anisotropic_refinement_flag() ||
+                       ((this_face->child(0)->vertex_index(3) ==
+                         this_face->child(1)->vertex_index(2)) &&
+                        (this_face->child(0)->vertex_index(3) ==
+                         this_face->child(2)->vertex_index(1)) &&
+                        (this_face->child(0)->vertex_index(3) ==
+                         this_face->child(3)->vertex_index(0))),
+                     ExcInternalError());
+
+              // DoFs on vertices:
+              // Nedelec elements have no DoFs on the vertices
+
+              // DoFs on lines:
+              // the DoFs on the interior lines to the children; the order
+              // of these lines is shown above (see n_intern_lines_on_children)
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(0)->line(1)->dof_index(dof, fe_index));
+
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(2)->line(1)->dof_index(dof, fe_index));
+
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(0)->line(3)->dof_index(dof, fe_index));
+
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(1)->line(3)->dof_index(dof, fe_index));
+
+              // DoFs on the bordering lines:
+              // DoFs on the exterior lines to the children; the order of
+              // these lines is shown above (see n_external_lines_on_children)
+              for (unsigned int line = 0; line < 4; ++line)
+                for (unsigned int child = 0; child < 2; ++child)
+                  for (unsigned int dof = 0; dof != fe.n_dofs_per_line(); ++dof)
```

Comment:
```suggestion
                  for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
```
Is there a reason for using a different style here?